lib/kino/process.ex

```diff
@@ -137,6 +139,212 @@ defmodule Kino.Process do
+          100 * thing
+        end,
+        max_concurrency: 3)
+        |> Stream.run()
+
+        Agent.stop(agent_pid)
+      end)
+  """
+  @spec seq_trace(trace_target(), (() -> any())) :: Markdown.t()
+  def seq_trace(pid, trace_function) when is_pid(pid) do
+    seq_trace([pid], trace_function)
+  end
+
+  def seq_trace(trace_pids, trace_function) do
+    # Set up the process message tracer and the Erlang seq_trace_module
+    {:ok, tracer_pid} = Tracer.start_link(nil)
+    :seq_trace.set_token(:send, true)
+    :seq_trace.set_token(:receive, true)
+    :seq_trace.set_token(:monotonic_timestamp, true)
+    previous_tracer = :seq_trace.set_system_tracer(tracer_pid)
+
+    # Run the user supplied function and capture the events if
+    # no errors were encountered
+    trace_events =
+      try do
+        # Run the user provided function
+        trace_function.()
+
+        # Extract all of the events from the system tracer
+        Tracer.get_trace_events(tracer_pid)
+      after
+        # The Tracer GenServer is no longer needed, shut it down
+        GenServer.stop(tracer_pid)
+
+        # Reset all of the tracing options
+        :seq_trace.set_system_tracer(previous_tracer)
+        :seq_trace.reset_trace()
+      end
+
+    # Get all of the events from the Tracer GenServer
+    trace_events =
+      trace_events
+      |> Enum.filter(fn
+        # Skip :spawn_reply messages
+        {_, _, _, _, {:spawn_reply, _, _, _}} ->
+          false
+
+        # Skip loopback messages
+        {_, _, pid, pid, _message} ->
+          false
+
+        # Filter out messages based on the trace target
+        {_, _, from_pid, to_pid, _message} ->
+          trace_pids == :all or from_pid in trace_pids or to_pid in trace_pids
+
+        # Reject the rest
+        _ ->
+          false
+      end)
+      |> Enum.sort_by(fn {_type, timestamp, _from, _to, _message} ->
+        timestamp
+      end)
+
+    # Get all the participating actors in the trace along with their sequence diagram IDs
+    {participants_lookup, _idx} =
+      trace_events
+      |> Enum.reduce({%{}, 0}, fn {_type, _timestamp, from, to, _message}, acc ->
+        acc
+        |> maybe_add_participant(from)
+        |> maybe_add_participant(to)
+      end)
+
+    # Generate the Mermaid formatted list of participants
+    participants =
+      participants_lookup
+      |> Enum.map_join("\n", fn {pid, idx} ->
+        case Process.info(pid, :registered_name) do
+          nil ->
+            "participant #{idx} AS #35;PID#{:erlang.pid_to_list(pid)};"
+
+          {:registered_name, []} ->
+            "participant #{idx} AS #35;PID#{:erlang.pid_to_list(pid)};"
+
+          {:registered_name, name} ->
+            "participant #{idx} AS #{inspect(name)};"
+        end
+      end)
+
+    # Generate the mermaid formatted list of message events
+    messages =
+      trace_events
+      |> Enum.map_join("\n", fn {type, _timestamp, from, to, message} ->
```

Comment:
It is a style thing but can you please remove the "single pipe" pipelines? More than one `|>` is fine tho. :)